pkg/ssh/ssh.go

```diff
@@ -502,42 +494,253 @@ func RunSSHUploadStakingFiles(host *models.Host, nodeInstanceDirPath string) err
+	bootstrapIDs, _ := utils.GetValueString(remoteAvagoConfFile, "bootstrap-ids")
+	bootstrapIPs, _ := utils.GetValueString(remoteAvagoConfFile, "bootstrap-ips")
+	avagoConf.BootstrapIDs = bootstrapIDs
+	avagoConf.BootstrapIPs = bootstrapIPs
+	// ready to render node config
+	nodeConf, err := remoteconfig.RenderAvalancheNodeConfig(avagoConf)
+	if err != nil {
 		return err
 	}
-	return docker.StartDockerComposeService(host, utils.GetRemoteComposeFile(), "avalanchego", constants.SSHLongRunningScriptTimeout)
+	if err := os.WriteFile(nodeConfFile.Name(), nodeConf, constants.WriteReadUserOnlyPerms); err != nil {
+		return err
+	}
+	return host.Upload(nodeConfFile.Name(), remoteconfig.GetRemoteAvalancheNodeConfig(), constants.SSHFileOpsTimeout)
 }
 
-// RunSSHUpdateSubnet runs avalanche subnet join <subnetName> in cloud server using update subnet info
-func RunSSHUpdateSubnet(host *models.Host, subnetName, importPath string) error {
-	if err := docker.StopDockerComposeService(host, utils.GetRemoteComposeFile(), "avalanchego", constants.SSHLongRunningScriptTimeout); err != nil {
+// RunSSHCreatePlugin runs script to create plugin
+func RunSSHCreatePlugin(host *models.Host, sc models.Sidecar) error {
+	vmID, err := sc.GetVMID()
+	if err != nil {
+		return err
+	}
+	subnetVMBinaryPath := fmt.Sprintf(constants.CloudNodeSubnetEvmBinaryPath, vmID)
+	hostInstaller := NewHostInstaller(host)
+	tmpDir, err := host.CreateTempDir()
+	if err != nil {
 		return err
 	}
-	if _, err := host.Command(fmt.Sprintf("/home/ubuntu/bin/avalanche subnet import file %s --force", importPath), nil, constants.SSHScriptTimeout); err != nil {
+	defer func(h *models.Host) {
+		_ = h.Remove(tmpDir, true)
+	}(host)
+	switch {
+	case sc.VM == models.CustomVM:
+		ux.Logger.Info("Building Custom VM for %s to %s", host.NodeID, subnetVMBinaryPath)
+		ux.Logger.Info("Custom VM Params: repo %s branch %s via %s", sc.CustomVMRepoURL, sc.CustomVMBranch, sc.CustomVMBuildScript)
+		if err := RunOverSSH(
+			"Build CustomVM",
+			host,
+			constants.SSHLongRunningScriptTimeout,
+			"shell/buildCustomVM.sh",
+			scriptInputs{
+				CustomVMRepoDir:     tmpDir,
+				CustomVMRepoURL:     sc.CustomVMRepoURL,
+				CustomVMBranch:      sc.CustomVMBranch,
+				CustomVMBuildScript: sc.CustomVMBuildScript,
+				VMBinaryPath:        subnetVMBinaryPath,
+			},
+		); err != nil {
+			return err
+		}
+
+	case sc.VM == models.SubnetEvm:
+		ux.Logger.Info("Installing Subnet EVM for %s", host.NodeID)
+		dl := binutils.NewSubnetEVMDownloader()
+		installURL, _, err := dl.GetDownloadURL(sc.VMVersion, hostInstaller) // extension is tar.gz
+		if err != nil {
+			return err
+		}
+
+		archiveName := "subnet-evm.tar.gz"
+		archiveFullPath := filepath.Join(tmpDir, archiveName)
+
+		// download and install subnet evm
+		if _, err := host.Command(fmt.Sprintf("%s %s -O %s", "busybox wget", installURL, archiveFullPath), nil, constants.SSHLongRunningScriptTimeout); err != nil {
```

Comment:
I believe you have scripts for this on shell, eg upgradeSubnetEVM, pkg/ssh/shell/getNewSubnetEVMRelease.sh, why not using them?

Reply:
why not having one .sh file for all of this?

Reply:
it will increase size of `scriptInputs` struct so I though we can get away without it as operation is straightforward